crates/wit-parser/src/decoding.rs

```diff
@@ -726,6 +727,40 @@ impl WitPackageDecoder<'_> {
         Ok(())
     }
 
+    fn register_export(
```

Comment:
I think this function may want to get folded into `register_import` perhaps? (maybe coupled with a function rename). The internal structure looks quite similar and I think it's needed for importing an interface with `nest` as well?